spec/datadog/appsec/actions_handler_spec.rb

```diff
@@ -97,4 +101,124 @@
       end
     end
   end
+
+  describe '.generate_stack' do
+    let(:generate_stack_action) { { 'stack_id' => 'foo' } }
+    let(:trace_op) { Datadog::Tracing::TraceOperation.new }
+    let(:trace_op_metastruct) { nil }
+    let(:span_op) { Datadog::Tracing::SpanOperation.new('span_test') }
+    let(:span_op_metastruct) { nil }
+    let(:context) { OpenStruct.new(trace: trace_op, span: span_op) }
+    let(:stack_trace_enabled) { true }
+    let(:max_collect) { 0 }
+
+    let(:stack_key) { Datadog::AppSec::Ext::TAG_STACK_TRACE }
+    let(:exploit_category) { Datadog::AppSec::Ext::EXPLOIT_PREVENTION_EVENT_CATEGORY }
+
+    before do
+      Datadog.configure do |c|
+        c.appsec.stack_trace.max_depth = 0
+        c.appsec.stack_trace.max_depth_top_percent = 0
+      end
+      allow(Datadog.configuration.appsec.stack_trace).to receive(:enabled).and_return(stack_trace_enabled)
+      allow(Datadog.configuration.appsec.stack_trace).to receive(:max_collect).and_return(max_collect)
+      allow(Datadog::AppSec).to receive(:active_context).and_return(context)
+      allow(Datadog::AppSec::ActionsHandler::StackTraceCollection).to receive(:collect).and_return(
+        ThreadBacktraceHelper.locations_inside_nested_blocks.map.with_index do |location, index|
+          {
+            id: index,
+            text: location.to_s.encode('UTF-8'),
+            file: (location.absolute_path || location.path)&.encode('UTF-8'),
+            line: location.lineno,
+            function: location.label&.encode('UTF-8')
+          }
+        end
```

Comment:
TBH i thought that helper suppose to make things easier. I think here it's either missing something or else ...